src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = item[key];
+
+    if (!result[<string>value]) {
+      result[<string>value] = [];
+    }
```

Comment:
```suggestion
    const value = String(item[key]);

    if (!result[value]) {
      result[value] = [];
    }
```
better to make string one time